packages/fxa-settings/src/lib/integrations/integration-factory.ts

```diff
@@ -2,128 +2,291 @@
+      // resumed.
+      const resume = integration.data.get('resume');
 
-  //   const verificationInfo = new SameBrowserVerificationModel(
-  //     {},
-  //     {
-  //       email: urlVerificationInfo.email,
-  //       namespace: namespace,
-  //       uid: urlVerificationInfo.uid,
-  //     }
-  //   );
-  //   verificationInfo.load();
+      if (!!resume) {
+        // Since we are resuming a previous flow, check to see if a previous oauth state was saved prior
+        // to the verification email being sent out and attempt to restore this state.
+        //
+        // If a state is present, the it will contain 'scope' and 'state' parameters that allow us to
+        // redirect the user back to the relying party's site after they complete their workflow here.
+        //
+        // If the state isn't present, we can still proceed with the user flow, but we will not be able to
+        // redirect the user back to the relying parties site, becuase without the knowing the relying party's
+        // oauth 'state', the redirect would be rejected.
+        integration.restoreOAuthState();
+      }
+    } else {
+      // Sign inflow
+      // params listed in:
+      // https://mozilla.github.io/ecosystem-platform/api#tag/OAuth-Server-API-Overview
+    }
+  }
 
-  //   return verificationInfo;
-  // }
+  initClientInfo(integration: OAuthIntegration) {
+    integration.clientInfo = this.createClientInfo(integration.data.clientId);
+  }
 
-  private createOAuthIntegration(relier: OAuthRelier) {
-    const flags = this.flags;
+  initSubscriptionInfo(integration: Integration) {
+    // Do not wait on this. Components can do so with useEffect if needed. However,
+    // not all
+    integration.subscriptionInfo = this.createRelierSubscriptionInfo();
+  }
 
-    if (flags.isDevicePairingAsAuthority()) {
-      return new PairingAuthorityIntegration();
+  private async createClientInfo(clientId: string | undefined) {
+    // Make sure a valid client id is provided before evening attempting the call.
+    if (!clientId) {
+      throw new OAuthError('UNKNOWN_CLIENT', {
+        client_id: 'null or empty',
+      });
     }
 
-    if (flags.isDevicePairingAsSupplicant()) {
-      return new PairingSupplicantIntegration(relier);
+    try {
+      const serviceInfo = await this.delegates.getClientInfo(clientId);
+      const clientInfo = new ClientInfo(new GenericData(serviceInfo));
+      return clientInfo;
+    } catch (err) {
+      if (
+        err.name === 'INVALID_PARAMETER' &&
+        err.validation?.keys?.[0] === 'client_id'
+      ) {
+        throw new OAuthError('UNKNOWN_CLIENT', {
+          client_id: clientId,
+        });
+      }
+
+      throw err;
     }
+  }
 
-    return new OAuthIntegration(relier);
+  private async createRelierSubscriptionInfo(): Promise<RelierSubscriptionInfo> {
+    // TODO: Is the following still needed? Seems like there should be a cleaner way to do this.
```

Comment:
Similar thing happens with clientInfo. I don't love it, and it has the side effect that initialization become sort of async. I wonder if we can tease client info and subscription info into their own modules, so they aren't part of the integration? They feel more like API calls than state ayways. I bet this is easier said than done, but just a thought.

Reply:
That sounds good to me. I reference FXA-6836 above where we set `const delegates = {` in `useIntegration` - I think we can create a hook for this instead and move it out. I can leave a comment on that ticket and bump it to 3 points instead of 2.